src/Lifetime/Managers/ExternallyControlledLifetimeManager.cs

```diff
@@ -27,14 +27,14 @@ public class ExternallyControlledLifetimeManager : LifetimeManager,
         /// <returns>the object desired, or null if no such object is currently stored.</returns>
         public override object GetValue(ILifetimeContainer container = null)
         {
-            if (null == _value) return NoValue;
-
-            var target = _value.Target;
-            if (_value.IsAlive) return target;
-
-            _value = null;
-
-            return NoValue;
+            var value = _value?.Target;
+            if (value == null) 
```

Comment:
Theoretically speaking, `null could be a valid value.

Reply:
How would that work? There's no way to detect if the WeakReference was created with null or with an object that has been garbage collected. You'd need some other state in ExternallyControlledLifetimeManager to track that the actual value is null.